edx-olive/edx-platform
# SE-1151 Inlines all the discussion underscore templates

This change reduces IO load on the edxapp servers by combining the 34 separate underscore templates used by the Discussion XBlock into a single file.

**Testing instructions**
Automated tests should verify main functionality.

To test manually:

1. Create a course with an inline discussion block.
1. Ensure that all the buttons and functions available to the block still work.
1. Use the main discussion forum, and ensure the same.

**Author Notes & Concerns**

Any changes not directly related to this fix were done to make the tests pass again.  Most of it doesn't need upstreaming (either it already is, or the file doesn't exist anymore in master), but:

* code changes for e235766 and 8029b5b also need to be upstreamed.
  These changes to the models are required to ensure that running the platform with a non-english language as the default don't alter migrations.
  (The migration changes for these commits were only added to this fork, so don't need upstreaming.)

**Reviewer**
